contracts/InterchainTokenService.sol

```diff
@@ -387,6 +417,9 @@ contract InterchainTokenService is
         } else if (trustedAddressHash(chainName()) == ITS_HUB_ROUTING_IDENTIFIER_HASH) {
             // Currently, deployments directly on ITS contract (instead of ITS Factory) are restricted for ITS contracts deployed on Amplifier, i.e registered with the Hub
             revert NotSupported();
+        } else if (bytes(destinationChain).length == 0) {
```

Comment:
```suggestion
        } else if (bytes(destinationChain).length == 0) {
```
We can collapse this check since Amplifier ITS chains never had local tokens deployed via this method.

Reply:
Yeah, there's already a check to prevent self deployment which should be enough?
`if (chainNameHash == keccak256(bytes(destinationChain))) revert CannotDeployRemotelyToSelf();`